File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <cmath>
#include <algorithm>
#include <iostream>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    
    string numStr = to_string(n);
    vector<char> element;
    for(int i=numStr.length() - 1; i>=0; --i)
    {
        element.push_back(numStr[i]);
    }
    sort(element.begin(), element.end());
    for(int i=0; i<element.size(); ++i)
    {
        answer += (element[i] - '0') * pow(10, i);
    }
    
    return answer;
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cs
using System;

public class Solution {
    public long solution(long n) {
        long answer = 0;
        
        string str = n.ToString();
        char[] cArr = new char[str.Length];
        for(int i = 0; i < str.Length; ++i)
            cArr[i] = str[i];
        Array.Sort(cArr);
        Array.Reverse(cArr);
        for(int i = 0; i < cArr.Length; ++i)
            Console.Write(cArr[i]);
        
        for(int i = 0; i < str.Length; ++i)
            answer = 10 * answer + cArr[i] - '0';
        
        return answer;
    }
}